feat(copy): improve autocapture and custom insight date range copy

thmsobrmlr opened the pull request:
## Problem

The autocapture description in settings [isn't accurate](https://posthog.slack.com/archives/C02E3BKC78F/p1690448666026269) and the tooltip for [custom insight date range isn't clear](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1690394290863519).

## Changes

This PR changes that.

## How did you test this code?

No tests.

thmsobrmlr committed:
update autocapture description in settings to reflect reality

thmsobrmlr committed:
make tooltip for custom date range more clear

Twixes committed:
Tighten autocapture copy

Twixes commented on frontend/src/lib/components/DateFilter/DateFilter.tsx:
```diff
@@ -101,7 +101,7 @@ export function DateFilter({
                         values[0],
                         values[1],
-                        'No date selected',
+                        'Date range specified in the respective Insight',
```
The tweaks around autocapture look good (I pushed a couple more though, so let me know if that makes sense), but I don't _think_ this solves the issue with the dashboard time range selector being confusing. There, the problem seems to be "Custom" being shown, and not "No date selected" (which I actually don't know how to get to show up anywhere)

thmsobrmlr replied:
Cool, thanks for the changes! Didn't want to change the "Custom" label here as it looks like a more complicated change. Will do this in a follow up PR.

thmsobrmlr replied:
https://github.com/PostHog/posthog/pull/16870

Twixes replied:
It definitely. I really don't know where this "No date selected"/"Date range specified in the respective Insight" was shown at all, do you have any pointers how to check this out?

thmsobrmlr replied:
Ah, got you. This is the tooltip for the "Custom" option in the dashboard date range selector.
<img width="434" alt="Screenshot 2023-08-02 at 01 38 43" src="https://github.com/PostHog/posthog/assets/1851359/3aa8fe58-18ad-401b-b451-aadebe9443f9">

Twixes replied:
Ohh, I see, didn't notice  it before

Twixes committed:
Lowercase "insight"

"Insights" as a feature is a proper noun, but we're talking about the entities and not the feature here.
